src/app/extensions/wishlists/shared/select-wishlist-modal/select-wishlist-modal.component.ts

```diff
@@ -40,141 +42,181 @@ export class SelectWishlistModalComponent implements OnInit, OnDestroy {
    */
   @Output() submitEmitter = new EventEmitter<{ id: string; title: string }>();
 
-  preferredWishlist: Wishlist;
-  updateWishlistForm: FormGroup;
-  wishlistOptions: SelectOption[];
+  wishlistOptions$: Observable<SelectOption[]>;
+
+  formGroup: FormGroup = new FormGroup({});
+  model: { wishlist?: string; newList?: string } = {};
+  multipleFieldConfig$: Observable<FormlyFieldConfig[]>;
+  singleFieldConfig: FormlyFieldConfig[];
 
   showForm: boolean;
-  newWishlistInitValue = '';
 
   modal: NgbModalRef;
 
-  idAfterCreate = '';
   private destroy$ = new Subject<void>();
 
   @ViewChild('modal') modalTemplate: TemplateRef<unknown>;
 
   constructor(
     private ngbModal: NgbModal,
-    private fb: FormBuilder,
-    private translate: TranslateService,
-    private wishlistsFacade: WishlistsFacade
+    private wishlistsFacade: WishlistsFacade,
+    private translate: TranslateService
   ) {}
 
   ngOnInit() {
-    this.wishlistsFacade.preferredWishlist$
-      .pipe(take(1), takeUntil(this.destroy$))
-      .subscribe(wishlist => (this.preferredWishlist = wishlist));
-    this.determineSelectOptions();
-    this.formInit();
-    this.wishlistsFacade.currentWishlist$
-      .pipe(takeUntil(this.destroy$))
-      .subscribe(wishlist => (this.idAfterCreate = wishlist && wishlist.id));
-
-    this.translate
-      .get('account.wishlists.choose_wishlist.new_wishlist_name.initial_value')
-      .pipe(take(1), takeUntil(this.destroy$))
-      .subscribe(res => {
-        this.newWishlistInitValue = res;
-        this.setDefaultFormValues();
-      });
-    this.updateWishlistForm.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(changes => {
-      if (changes.wishlist !== 'newList') {
-        this.updateWishlistForm.get('newWishlist').clearValidators();
-      } else {
-        this.updateWishlistForm.get('newWishlist').setValidators(Validators.required);
-      }
-      this.updateWishlistForm.get('newWishlist').updateValueAndValidity({ emitEvent: false });
-    });
-  }
-
-  ngOnDestroy() {
-    this.destroy$.next();
-    this.destroy$.complete();
-  }
-
-  private formInit() {
-    this.updateWishlistForm = this.fb.group({
-      wishlist: [
-        this.wishlistOptions && this.wishlistOptions.length > 0 ? this.wishlistOptions[0].value : 'newList',
-        Validators.required,
-      ],
-      newWishlist: [this.newWishlistInitValue, Validators.required],
-    });
-  }
+    this.singleFieldConfig = [
```

Comment:
maybe a comment about the blocks would be helpful. What is meant with singleFieldConfig,  wishlistOptions and multifieldconfig?